extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/utils.js

```diff
@@ -0,0 +1,463 @@
+    target.style.msFilter = "progid:DXImageTransform.Microsoft.Dropshadow(OffX=0, OffY=0, Color='#fff')";
+  },
+
+  /**
+  * Remove the highlight effect of the given element
+  *
+  * @function removeHighlight
+  * @param {DOMElement} target - The element to remove the highlight effect from
+  * @param {DOMElement} opacity - The opacity to return the element to
+  */
+  removeHighlight: function(target, opacity, color) {
+    target.style.opacity = opacity;
+    target.style.color = color;
+    target.style.WebkitFilter = "";
+    target.style.filter = "";
+    target.style.msFilter = "";
+  },
+
+  /**
+  * Determine which buttons should be shown in the control bar given the width of the player<br/>
+  * Note: items which do not meet the item spec will be removed and not appear in the results.
+  *
+  * @function collapse
+  * @param {Number} barWidth - Width of the control bar
+  * @param {Object[]} orderedItems - array of left to right ordered items. Each item meets the skin's "button" schema.
+  * @returns {Object} An object of the structure {fit:[], overflow:[]} where the fit object is
+  *   an array of buttons that fit in the control bar and overflow are the ones that should be hidden
+  */
+  collapse: function( barWidth, orderedItems, responsiveUIMultiple ) {
+    if( isNaN( barWidth ) || barWidth === undefined ) { return orderedItems; }
+    if( ! orderedItems ) { return []; }
+    var self = this;
+    var validItems = orderedItems.filter( function(item) { return self._isValid(item); } );
+    var r = this._collapse( barWidth, validItems, responsiveUIMultiple );
+    return r;
+  },
+
+  /**
+  * Find thumbnail image URL and its index that correspond to given time value
+  *
+  * @function findThumbnail
+  * @param {Object} thumbnails - metadata object containing information about thumbnails
+  * @param {Number} hoverTime - time value to find thumbnail for
+  * @param {Number} duration - duration of the video
+  * @returns {Object} object that contains URL and index of requested thumbnail
+  */
+  findThumbnail: function(thumbnails, hoverTime, duration) {
+    var timeSlices = thumbnails.data.available_time_slices;
+    var width = thumbnails.data.available_widths[0]; //choosing the lowest size
+
+    var position = Math.floor((hoverTime/duration) * timeSlices.length);
+    position = Math.min(position, timeSlices.length - 1);
+    position = Math.max(position, 0);
+
+    var selectedTimeSlice = null;
+    var selectedPosition = position;
+
+    if (timeSlices[position] >= hoverTime) {
+      selectedTimeSlice = timeSlices[0];
+      for (var i = position; i >= 0; i--) {
+        if (timeSlices[i] <= hoverTime) {
+          selectedTimeSlice = timeSlices[i];
+          selectedPosition = i;
+          break;
+        }
+      }
+    } else {
+      selectedTimeSlice = timeSlices[timeSlices.length - 1];
+      for (var i = position; i < timeSlices.length; i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "i" as this name is already used in declaration at line 363. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of "break" and "continue" statements in this loop to use one at most. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ATooManyBreakOrContinueInLoop)